reactors/1D_packbed.ipynb

```diff
@@ -0,0 +1,518 @@
+    "n_gas = gas.n_species  # number of gas species\n",
+    "n_surf = surf.n_species  # number of surface species\n",
+    "n_gas_reactions = gas.n_reactions  # number of gas-phase reactions\n",
+    "\n",
+    "# Set offsets\n",
+    "offset_rhou = 0\n",
+    "offset_p = 1\n",
+    "offset_T = 2\n",
+    "offset_Y = 3\n",
+    "offset_Z = offset_Y + n_gas\n",
+    "n_var = offset_Z + n_surf  # total number of variables (rhou, P, T, Yk and Zk)\n",
+    "\n",
+    "print(\"Number of gas-phase species = \", n_gas)\n",
+    "print(\"Number of surface-phase species = \", n_surf)\n",
+    "print(\"Number of variables = \", n_var)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define reactor geometry and operating conditions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Modeling packed-bed, catalytic-membrane reactor...\n",
+      "H2 permeable membrane is present.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Reactor geometry\n",
+    "L = 5e-2  # length of the reactor (m)\n",
+    "R = 5e-3  # radius of the reactor channel (m)\n",
+    "phi = 0.5  # porosity of the bed (-)\n",
+    "tau = 2.0  # toruosity of the bed (-)\n",
+    "dp = 3.37e-4  # particle diameter (m)\n",
+    "As = 3.5e6  # specific surface area (1/m)\n",
+    "\n",
+    "# Energy (adiabatic or isothermal)\n",
+    "solve_energy = True  # True: Adiabatic, False: isothermal\n",
+    "\n",
+    "# Membrane\n",
+    "membrane_present = (\n",
+    "    True  # flag to include membrane (False = no membrane, True = membrane)\n",
+    ")\n",
+    "membrane_perm = 1e-15  # membrane permeability (kmol*m3/s/Pa)\n",
+    "thickness = 3e-6  # membrane thickness (m)\n",
+    "membrane_sp_name = \"H2\"  # membrane-permeable species name\n",
+    "p_sweep = 1e5  # partial pressure of permeable species in the sweep channel (Pa)\n",
+    "permeance = membrane_perm / thickness  # permeance of the membrane (kmol*m2/s/Pa)\n",
+    "\n",
+    "if membrane_present == True:\n",
+    "    print(\"Modeling packed-bed, catalytic-membrane reactor...\")\n",
+    "    print(membrane_sp_name, \"permeable membrane is present.\")\n",
+    "\n",
+    "# Get required properties based on the geometry and mechanism\n",
+    "W_g = gas.molecular_weights  # vector of molecular weight of gas species\n",
+    "vol_ratio = phi / (1 - phi)\n",
+    "eff_factor = phi / tau  # effective factor for permeability calculation\n",
+    "B_g = (\n",
+    "    pow(vol_ratio, 2) * pow(dp, 2) * eff_factor / 72\n",
+    ")  # permeability based on Kozeny-Carman equation\n",
+    "area2vol = 2 / R  # area to volume ratio assuming a cylindrical reactor\n",
+    "D_h = 2 * R  # hydraulic diameter\n",
+    "membrane_sp_ind = gas.species_index(\n",
+    "    membrane_sp_name\n",
+    ")  # index of the membrane-permeable species\n",
```

Comment:
```suggestion
    "membrane_sp_ind = gas.species_index(membrane_sp_name)\n",
```
This feels like it doesn't need a comment...

Reply:
Updated.